File: java/src/main/java/com/experian/dvs/client/address/suggestions/format/SuggestionsFormatResult.java
package com.experian.dvs.client.address.suggestions.format;

import com.experian.dvs.client.address.AddressConfidence;
import com.experian.dvs.client.server.address.suggestions.RestApiSuggestionsFormatResponse;
import com.experian.dvs.client.server.address.suggestions.RestApiSuggestionsFormatResult;

import java.util.List;

public class SuggestionsFormatResult {

    private final boolean moreResultsAvailable;
    private final AddressConfidence confidence;
    private final List<SuggestionsFormatSuggestion> suggestions;

    public SuggestionsFormatResult(RestApiSuggestionsFormatResponse response) {
        final RestApiSuggestionsFormatResult result = response.getResult();
        if (result != null) {
            this.moreResultsAvailable = Boolean.TRUE.equals(result.isMoreResultsAvailable());
            this.confidence = result.getConfidence() != null ? AddressConfidence.fromName(result.getConfidence()) : null;
            this.suggestions = result.getSuggestions() != null ? result.getSuggestions().stream().map(SuggestionsFormatSuggestion::new).toList() : List.of();
        } else {
            this.moreResultsAvailable = false;
            this.confidence = null;
            this.suggestions = List.of();
        }
    }

    public boolean isMoreResultsAvailable() {
        return moreResultsAvailable;
    }

    public AddressConfidence getConfidence() {
        return confidence;
    }

    public List<SuggestionsFormatSuggestion> getSuggestions() {
        return suggestions;
    }
}

File: csharp/DVSClient/DVSClient/address/suggestions/SuggestionsFormatResult.cs
using DVSClient.Common;
using DVSClient.Server.Address.Suggestions;

namespace DVSClient.Address.Suggestions
{
    public class SuggestionsFormatResult
    {
        public bool MoreResultsAvailable { get; }
        public AddressConfidence? Confidence { get; }
        public IEnumerable<SuggestionsFormatSuggestion> Suggestions { get; }

        public SuggestionsFormatResult(RestApiSuggestionsFormatResponse response)
        {
            var result = response.Result;
            if (result != null)
            {
                MoreResultsAvailable = result.MoreResultsAvailable;
                Confidence = result.Confidence?.GetEnumValueFromJsonName<AddressConfidence>();
                Suggestions = result.Suggestions?.Select(s => new SuggestionsFormatSuggestion(s)).ToList() ?? new List<SuggestionsFormatSuggestion>();
            }
            else
            {
                MoreResultsAvailable = false;
                Confidence = null;
                Suggestions = new List<SuggestionsFormatSuggestion>();
            }
        }
    }
}